src/main/java/edu/ucla/library/libcal/handlers/ProxyHandler.java

```diff
@@ -0,0 +1,121 @@
+
+package edu.ucla.library.libcal.handlers;
+
+import static edu.ucla.library.libcal.MediaType.APPLICATION_JSON;
+import static edu.ucla.library.libcal.MediaType.TEXT_PLAIN;
+import static info.freelibrary.util.Constants.EMPTY;
+import static info.freelibrary.util.Constants.SLASH;
+
+import info.freelibrary.util.HTTP;
+
+import edu.ucla.library.libcal.Constants;
+import edu.ucla.library.libcal.JsonKeys;
+import edu.ucla.library.libcal.MessageCodes;
+import edu.ucla.library.libcal.services.LibCalProxyService;
+import edu.ucla.library.libcal.services.OAuthTokenService;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A handler that processes status information requests.
+ */
+@SuppressWarnings("PMD.AvoidCatchingGenericException")
+public class ProxyHandler implements Handler<RoutingContext> {
+
+    /**
+     * The handler's logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProxyHandler.class, MessageCodes.BUNDLE);
+
+    /**
+     * A constant for evaluating request path.
+     */
+    private static final String LIBCAL = "libcal";
+
+    /**
+     * The handler's copy of the Vert.x instance.
+     */
+    private final Vertx myVertx;
+
+    /**
+     * A service for LibCal API calls
+     */
+    private final LibCalProxyService myApiProxy;
+
+    /**
+     * A service for LibCal OAuth calls
+     */
+    private final OAuthTokenService myTokenProxy;
+
+    /**
+     * Creates a handler that returns a status response.
+     *
+     * @param aVertx A Vert.x instance
+     */
+    public ProxyHandler(final Vertx aVertx) {
+        myVertx = aVertx;
+        myApiProxy = LibCalProxyService.createProxy(myVertx);
+        myTokenProxy = OAuthTokenService.createProxy(myVertx);
+    }
+
+    @Override
+    public void handle(final RoutingContext aContext) {
+        final HttpServerResponse response = aContext.response();
+        final String path = aContext.request().path();
+
+        try {
+            if (path.endsWith(LIBCAL) || path.endsWith(LIBCAL.concat(SLASH))) {
+                response.setStatusCode(HTTP.BAD_REQUEST).putHeader(HttpHeaders.CONTENT_TYPE, TEXT_PLAIN.toString())
+                        .end(LOGGER.getMessage(MessageCodes.LCP_006));
+            } else {
+                final String receivedQuery = path.replaceAll(LIBCAL.concat(SLASH), "")
+                        .concat(aContext.request().query() != null ? aContext.request().query() : "");
+                myTokenProxy.getBearerToken().compose(token -> {
+                    return myApiProxy.getLibCalOutput(token, SLASH.concat(receivedQuery)).onSuccess(apiOutput -> {
+                        response.setStatusCode(HTTP.OK).putHeader(HttpHeaders.CONTENT_TYPE,
```

Comment:
I changed `getLibCalOutput` to `Future<HttpResponse<String>> getLibCalOutput(String anOAuthToken, String aQuery)`... and the code generator really, **_really_** did not like it:
```
Jun 13, 2022 9:24:31 PM io.vertx.codegen.CodeGenProcessor reportGenException
SEVERE: Could not generate model for edu.ucla.library.libcal.services.LibCalProxyService: class io.vertx.codegen.type.ParameterizedTypeInfo cannot be cast to class io.vertx.codegen.type.ApiTypeInfo (io.vertx.codegen.type.ParameterizedTypeInfo and io.vertx.codegen.type.ApiTypeInfo are in unnamed module of loader java.net.URLClassLoader @53dd29e0)
io.vertx.codegen.GenException: class io.vertx.codegen.type.ParameterizedTypeInfo cannot be cast to class io.vertx.codegen.type.ApiTypeInfo (io.vertx.codegen.type.ParameterizedTypeInfo and io.vertx.codegen.type.ApiTypeInfo are in unnamed module of loader java.net.URLClassLoader @53dd29e0)
Caused by: java.lang.ClassCastException: class io.vertx.codegen.type.ParameterizedTypeInfo cannot be cast to class io.vertx.codegen.type.ApiTypeInfo (io.vertx.codegen.type.ParameterizedTypeInfo and io.vertx.codegen.type.ApiTypeInfo are in unnamed module of loader java.net.URLClassLoader @53dd29e0)
```

Reply:
Though the error message is different than what I would have expected (I've seen a different message before on an older version of Vert.x) it might be because service proxies have restrictions on the type of things they can return: https://vertx.io/docs/vertx-service-proxy/java/#_restrictions_for_service_interface. I don't think HttpResponse implements @DataObject which would be the only way it could be returned.

Reply:
Oh shoot, I forgot about that. Hmm...

Reply:
Could we have it return `Future<JsonObject>` instead, with something this wrapped in the future?
```
{ "body": ... , "statusCode": ... }
```

Reply:
If it were in the handler instead (and the handler had its own WebClient instance), it would be possible to get a direct reference to an HttpResponse object, since the message bus (which restricts the types of messages) wouldn't be involved.

Reply:
But then we're coupling the handler and the "service"... we did that a lot in our earlier Vert.x projects, but thought we were moving away from that.

Reply:
Yeah, I agree.

Reply:
Modified service to return JSON, with fields for status, status message, and body.